src/Compilers/Core/Portable/Compilation/Compilation.cs

```diff
@@ -1609,21 +1612,14 @@ internal void EnsureAnonymousTypeTemplates(CancellationToken cancellationToken)
             CompilationTestData testData,
             CancellationToken cancellationToken);
 
-        internal EmitResult Emit(
-            EmitStreamProvider peStreamProvider,
-            EmitStreamProvider pdbStreamProvider,
-            EmitStreamProvider xmlDocumentationStreamProvider,
-            EmitStreamProvider win32ResourcesStreamProvider,
+        internal CommonPEModuleBuilder PreCompile(
```

Comment:
It would be nice to have some documentation on the relative role of PreCompile, Compile, and PostCompile. Just a sentence of two on each, please.